docs/docs/building-apps/00-app-go.md

```diff
@@ -2,25 +2,129 @@
+Learn more about the rationale of App Wiring in [ADR-057](../architecture/adr-057-app-wiring.md).
+
+:::
+
+:::note
+
+### Pre-requisite Readings
+
+* [ADR 057: App Wiring](../architecture/adr-057-app-wiring.md)
+* [Depinject Documentation](../tooling/02-depinject.md)
+
+:::
+
+This section is intended to provide an overview of the `SimApp` `app.go` file with App Wiring.
+
+## `app_config.go`
+
+The `app_config.go` file is the single place to configure all modules parameters.
+
+1. Create the `AppConfig` variable:
+
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L77-L78
+    ```
+
+2. Configure the `runtime` module:
+
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L79-L137
+    ```
+
+3. Configure the modules defined in the `BeginBlocker` and `EndBlocker` and the `tx` module:
+
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L138-L156
+    ```
+
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L170-L173
+    ```
+
+### Complete `app_config.go`
+
+```go reference
+https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L52-L233
+```
+
+### Alternative formats
+
+:::tip
+The example above shows how to create an `AppConfig` using Go. However, it is also possible to create an `AppConfig` using YAML, or JSON.  
+The configuration can then be embed with `go:embed` and read with [`appconfig.LoadYAML`](https://pkg.go.dev/cosmossdk.io/core/appconfig#LoadYAML), or [`appconfig.LoadJSON`](https://pkg.go.dev/cosmossdk.io/core/appconfig#LoadJSON), in `app.go`.
+
+```go
+//go:embed app_config.yaml
+var (
+    appConfigYaml []byte
+    appConfig = appconfig.LoadYAML(appConfigYaml)
+)
+```
+
+:::
+
+```yaml
+modules:
+  - name: runtime
+    config:
+      "@type": cosmos.app.runtime.v1alpha1.Module
+      app_name: SimApp
+      begin_blockers: [staking, auth, bank]
+      end_blockers: [bank, auth, staking]
+      init_genesis: [bank, auth, staking]
+  - name: auth
+    config:
+      "@type": cosmos.auth.module.v1.Module
+      bech32_prefix: cosmos
+  - name: bank
+    config:
+      "@type": cosmos.bank.module.v1.Module
+  - name: staking
+    config:
+      "@type": cosmos.staking.module.v1.Module
+  - name: tx
+    config:
+      "@type": cosmos.tx.module.v1.Module
+```
+
+A more complete example of `app.yaml` can be found [here](https://github.com/cosmos/cosmos-sdk/blob/91b1d83f1339e235a1dfa929ecc00084101a19e3/simapp/app.yaml).
 
 ## `app.go`
 
-Since `v0.47.0` the Cosmos SDK have made easier wiring an `app.go` thanks to dependency injection:
+`app.go` is the place where `SimApp` is constructed. `depinject.Inject` facilitates that by automatically wiring the app modules and keepers, provided an application configuration, `AppConfig`. `SimApp` is constructed, when calling the injected `*runtime.AppBuilder`, with `appBuilder.Build(...)`.    
```

Comment:
> `depinject.Inject` facilitates that by automatically wiring the app modules and keepers, provided an application configuration, `AppConfig`

this sentence seems incomplete.  Maybe "provided an application configuration `AppConfig` is provided." ?